zkstack_cli/crates/zkstack/src/commands/dev/commands/track_priority_txs.rs

```diff
@@ -0,0 +1,284 @@
+use std::{sync::Arc, thread::sleep, time::Duration};
+
+use anyhow::Context;
+use chrono::{DateTime, Local};
+use clap::Parser;
+use cliclack::clear_screen;
+use ethers::{
+    providers::{Http, Middleware, Provider},
+    types::{Filter, H256},
+};
+use zkstack_cli_common::logger;
+use zksync_basic_types::Address;
+use zksync_contracts::hyperchain_contract;
+use zksync_types::l1::L1Tx;
+use zksync_web3_decl::{
+    client::{Client, L2},
+    namespaces::{EthNamespaceClient, ZksNamespaceClient},
+};
+
+use crate::{
+    consts::DEFAULT_EVENTS_BLOCK_RANGE,
+    utils::{
+        addresses::apply_l1_to_l2_alias,
+        providers::{get_ethers_provider, get_zk_client},
+    },
+};
+
+const DEFAULT_DISPLAYED_TXS_LIMIT: usize = 20;
+const DEFAULT_INTERVALS_SECS: u64 = 2;
+
+#[derive(Clone)]
+enum PriorityTxState {
+    Success,
+    Fail,
+    WaitingToProcess(u64),
+}
+
+#[derive(Clone)]
+struct TxStatus {
+    block_timestamp: u64,
+    eth_tx_hash: H256,
+    priority_tx_hash: H256,
+    state: PriorityTxState,
+}
+
+#[derive(Clone, Debug, Parser)]
+pub struct TrackPriorityOpsArgs {
+    #[clap(long)]
+    l2_chain_id: u64,
+    #[clap(long)]
+    l1_rpc_url: String,
+    #[clap(long)]
+    l2_rpc_url: String,
+    #[clap(long)]
+    l1_op_sender: Option<Address>,
+    #[clap(long)]
+    from_block: Option<u64>,
+    #[clap(long)]
+    limit: Option<usize>,
+    #[clap(long)]
+    watch: Option<bool>,
+    #[clap(long)]
+    update_frequency_ms: Option<u64>,
```

Comment:
can we add an option to specify tx_hash?

Reply:
okay